python/grass/gunittest/utils.py

```diff
@@ -80,3 +82,17 @@ def safe_repr(obj, short=False):
     if not short or len(result) < _MAX_LENGTH:
         return result
     return result[:_MAX_LENGTH] + " [truncated]..."
+
+
+def xfail_windows(test_item):
+    """Marks a test as an expected failure or error only on Windows
+    Equivalent to applying @unittest.expectedFailure only when running
+    on Windows.
+    """
+    if not sys.platform.startswith("win"):
+        return lambda func: func
+    warnings.warn(
+        "Once the test is fixed and passing, remove the @xfail_windows decorator",
+        stacklevel=2,
+    )
```

Comment:
Why do you issue a warning?

Reply:
As, contrary to the skipif mechanism, there really isn't any way to have a reason attached to an xfail. I was directly at the unittest source code to really find out for myself. I think the decorator by itself is not descriptive enough by itself, and don't want the xfail to stay forever (I know the saying that there's nothing more permanent than a temporary solution...). Using the warning mechanism, it is displayed at least once and there's a trace of what it is. Imagine when a test will be fixed. I want that to appear somewhere in the stack trace or code read to know what to do when a test is fixed.